vue/src/components/loki/LogLabelSelect.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <v-chip
+    v-model="isLabelSelected"
+    label
+    small
+    class="ma-1"
+    v-if="label"
+    :color="isLabelSelected ? 'blue' : 'grey lighten-4'"
+    :class="{ active: isLabelSelected }"
+    @click="setIsLabelSelected"
+  >
+    {{ label.name }}
+  </v-chip>
+</template>
+
+<script lang="ts">
+import { defineComponent, shallowRef, PropType, watch, computed } from '@vue/composition-api'
+
+// Composables
+import { UseDateRange } from '@/use/date-range'
+import { Label } from '@/components/loki/logql'
+
+export default defineComponent({
+  name: 'LogLabelSelect',
+  props: {
+    dateRange: {
+      type: Object as PropType<UseDateRange>,
+      required: true,
+    },
+    label: {
+      type: Object as PropType<Label>,
+      require: true,
+    },
```

Comment:
It is better to not pass objects in props, e.g. in this case you should have 2 separate props:

```js
label: {
    type: String,
    required: true,
},
selected: {
    type: Boolean,
    default: false,
}.
```

There are exception to that rule, but most of the time it is true.